servers/src/common/adapters.rs

```diff
@@ -498,14 +499,16 @@ impl NetToChainAdapter {
 
 	// Find the first locator hash that refers to a known header on our main chain.
 	fn find_common_header(&self, locator: &[Hash]) -> Option<BlockHeader> {
-		let txhashset = self.chain().txhashset();
-		let txhashset = txhashset.read();
+		let header_pmmr = self.chain().header_pmmr();
+		let header_pmmr = header_pmmr.read();
 
 		for hash in locator {
 			if let Ok(header) = self.chain().get_block_header(&hash) {
-				if let Ok(header_at_height) = txhashset.get_header_by_height(header.height) {
-					if header.hash() == header_at_height.hash() {
-						return Some(header);
+				if let Ok(hash_at_height) = header_pmmr.get_header_hash_by_height(header.height) {
+					if let Ok(header_at_height) = self.chain().get_block_header(&hash_at_height) {
+						if header.hash() == header_at_height.hash() {
```

Comment:
This check probably shouldn't be necessary anymore since we're retrieving by hash.

Reply:
Ahh looks like we can simplify this, but we do still need the check.
The core of this is to check the provided header against the header at that height on the current chain - if they have the same hash then the header is on the current chain.

1. look up the header based on hash
1. look up the header on our chain based on this height
1. check these are the same header by hash

But we don't need to actually look the header up based on `hash_at_height` as we can just compare this hash directly.

```
if let Ok(header) = self.chain().get_block_header(&hash) {
	if let Ok(hash_at_height) = header_pmmr.get_header_hash_by_height(header.height) {
		if header.hash() == hash_at_height {
			return Some(header);
		}
	}
}
```